### ISSUE TITLE
CreateProcess fails with ERROR_INVALID_PARAMETER when trying to run a file on mounted volume

### ISSUE THREAD

**Issue by Karandra** [dokan-dev/dokany#650]
# Environment
 * Windows version: Windows 7 x64 SP1
 * Processor architecture: x64
 * Dokany version:  2.0
 * Library type: Dokany
 
# Check List
  - [x] I checked my issue [doesn't exist yet](https://github.com/dokan-dev/dokany/issues?utf8=%E2%9C%93&q=is%3Aissue)
  - [x] My issue is valid with mirror default sample and not specific to my user-mode driver implementation
  - [x] I can always reproduce the issue with the provided description below.
  - [x] I have updated Dokany to the [latest version](https://github.com/dokan-dev/dokany/releases) and have reboot my computer after.
  - [ ] I tested one of the last [snapshot](https://github.com/dokan-dev/dokany/wiki/Build#user-snapshot) from appveyor CI

# Description
Using mirror sample I mounted file system to `M:` drive using entire game folder as root. FS mounted well and I tried to run file `enbhost.exe` there from explorer. Program does not start with error "The parameter is incorrect.". Then I tried to run the file using [CreateProcess](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682425(v=vs.85).aspx) function. It doesn't work either, CreateProcess failed and `GetLastError` returned `ERROR_INVALID_PARAMETER` (87).

Dokan 1 also has same issue, but only when I try to run programs in same process where Dokan is running, but I'm not sure if it's caused by Dokan or by my own FS implementation. For 2.0 I used mirror sample.

Why I can't run programs in mount point? Is it a Dokany bug or I doing something wrong? Here is the code I used, just in case.
```cpp
STARTUPINFO tStartupInfo = {0};
tStartupInfo.cb = sizeof(tStartupInfo);

PROCESS_INFORMATION tProcessInfo = {0};
tProcessInfo.hProcess = INVALID_HANDLE_VALUE;
tProcessInfo.hThread = INVALID_HANDLE_VALUE;

WCHAR s[1024] = L"M:\\enbhost.exe";
bool bOK = CreateProcessW
(
	s,
	NULL,
	NULL,
	NULL,
	FALSE,
	0,
	NULL,
	NULL,
	&tStartupInfo,
	&tProcessInfo
);
fprintf(stdout, "%s: %u\r\n", bOK ? "true" : "false", GetLastError());
```

# Logs
[DebugView++.txt](https://github.com/dokan-dev/dokany/files/1609124/DebugView.txt)

**Comment by Liryna** [comment]
@KerberX Hi,

Can you try with the mount manager option and see if it improves the situation?

**Comment by KerberX** [comment]
Hi, no, running as `mirror.exe /o /d /r "D:\Game Files\The Elder Scrolls\Skyrim" /l M:` doesn't change anything.

**Comment by KerberX** [comment]
@Liryna Hi,
I found why I can't run programs. This has something to do with GetFileSecurity function, because returning `STATUS_NOT_IMPLEMENTED` in MirrorGetFileSecurity solves the problem and allowing me to run any app.  I can't check it on Mirror sample (I can't compile it), but it works on my own implementation, whish is almost one-to-one copy of mirror.

This isn't good solution, GetFileSecurity can be important to others, but in my case it isn't. I don't really need security information for my purposes, at least right now. Although I'd like to know why MirrorGetFileSecurity prevents process creation.

**Comment by Liryna** [comment]
@KerberX Hi,

Well found 👍Probably there is a case not handle in MirrorGetFileSecurity that return INVALID_PARAMETER.

In your logs, there is an error "GetSecurityInfo error: 5" ERROR_ACCESS_DENIED probably releated to it.

I could not get the hand on my env for now.
an issue on the version 2.x.x, I made a fix not so long time ago but still another issue pending https://github.com/dokan-dev/dokany/issues/607 (during delete but should probably not be important for you)

**Comment by Liryna** [comment]
Hi @KerberX ,

Have you been able to get more information or a workaround about this issue ?

**Comment by KerberX** [comment]
Hi @Liryna ,
Except returning `STATUS_NOT_IMPLEMENTED ` I can't do anything to make it work. Sorry, I have not worked on this since my last reply.

**Comment by Liryna** [comment]
Hi @KerberX ,

Do you think there is a possibility you can do the test on Dokan version 1 ? Since v2 release there is a fix that could be what you are facing.

**Comment by KerberX** [comment]
@Liryna ,
I've tested on latest 1.x release (1.2.1.2000) on slightly modified mirror sample with same results. I pasted [this](https://github.com/KerberX/KxVirtualFileSystem/blob/master/KxVirtualFileSystem/KxVirtualFileSystemMain.cpp#L41-L75) part into mirror's `main` function and placed `DokanMain` call into another thread (using `std::thread` class).

I can run a program, but opening file browse dialog causes mirror to hung. Even worse, it can freeze entire system, v2 has some kind of timeout in such case, this version just freezes forever.
